apps/server/src/shared/domain/service/lesson-copy.service.ts

```diff
@@ -14,38 +14,61 @@ export class LessonCopyService {
 	constructor(private readonly copyHelperService: CopyHelperService) {}
 
 	copyLesson(params: LessonCopyParams): CopyStatus {
+		const { copiedContent, contentStatus } = this.copyLessonContents(params.originalLesson.contents || []);
 		const copy = new Lesson({
 			course: params.destinationCourse,
 			hidden: true,
 			name: this.copyHelperService.deriveCopyName(params.originalLesson.name),
 			position: params.originalLesson.position,
-			contents: this.copyTextComponent(params.originalLesson.contents || []),
+			contents: copiedContent,
 		});
 
-		const elements = [...this.lessonStatusElements(), ...this.textComponentStatus(copy.contents)];
+		const elements = [...this.lessonStatusMetadata(), ...contentStatus];
 
 		const status: CopyStatus = {
 			title: copy.name,
 			type: CopyElementType.LESSON,
-			status: CopyStatusEnum.SUCCESS,
+			status: this.copyHelperService.deriveStatusFromElements(elements),
 			copyEntity: copy,
 			elements,
 		};
 
 		return status;
 	}
 
-	private copyTextComponent(originalContent: IComponentProperties[]): IComponentProperties[] {
+	private copyLessonContents(contents: IComponentProperties[]) {
 		const copiedContent: IComponentProperties[] = [];
-		originalContent.forEach((element) => {
-			if (element.component === ComponentType.TEXT) {
+		const copiedContentStatus: CopyStatus[] = [];
+		contents.forEach((element) => {
+			if (element.component === ComponentType.TEXT || element.component === ComponentType.LERNSTORE) {
 				copiedContent.push(element);
+				copiedContentStatus.push({
+					title: element.title,
+					type: CopyElementType.LESSON_CONTENT,
+					status: CopyStatusEnum.SUCCESS,
+				});
+			} else if (element.component === ComponentType.GEOGEBRA) {
+				const geoGebraContent = this.copyGeogebra(element);
+				copiedContent.push(geoGebraContent);
+				copiedContentStatus.push({
+					title: element.title,
+					type: CopyElementType.LESSON_CONTENT,
+					status: CopyStatusEnum.PARTIAL,
+				});
 			}
 		});
-		return copiedContent;
+		const contentStatus = this.contentStatus(copiedContentStatus);
```

Comment:
I think naming the return value contentStatus and the function deriving this status with the same name is a bit missleading. maybe the private function "contentStatus" could be a bit more directly named on what it does "deriveContentStatus" or sth